File: backend/core/name_validation.py
import re

_NAME_PATTERN = re.compile(r"^[A-Za-z _]+$")

NAME_REQUIREMENTS = "Name may only contain letters, spaces, and underscores."


def validate_name(value: str) -> str:
    trimmed = value.strip()
    if not trimmed:
        raise ValueError("Name cannot be empty.")
    if not _NAME_PATTERN.match(trimmed):
        raise ValueError(NAME_REQUIREMENTS)
    if not re.search(r"[A-Za-z]", trimmed):
        raise ValueError("Name must contain at least one letter.")
    return trimmed

File: frontend/src/utils/nameValidation.ts
const NAME_PATTERN = /^[A-Za-z _]+$/;

export const NAME_REQUIREMENTS =
  "Name may only contain letters, spaces, and underscores.";

export function isValidName(name: string): boolean {
  const trimmed = name.trim();
  if (!trimmed) return false;
  if (!NAME_PATTERN.test(trimmed)) return false;
  return /[A-Za-z]/.test(trimmed);
}

export function normalizeName(name: string): string {
  return name.trim();
}
